Support for dataclass and attrs items (#3881)
scrapy/scrapy: docs/topics/exporters.rst

      value unchanged except for ``unicode`` values which are encoded to
      ``str`` using the encoding declared in the :attr:`encoding` attribute.

      :param field: the field being serialized. If the source :ref:`item object
          <item-types>` does not define field metadata, *field* is an empty
          :class:`dict`.

>       :type field: :class:`~scrapy.item.Field` object or an empty :class:`dict`

Reviewer: ```suggestion
      :type field: :class:`~scrapy.item.Field` object or a :class:`dict` instance
```

Reviewer: the dict doesn't have to be empty, as we support meta for attr.s and dataclasses
